Tasks/Aleksiutovich/Task7/logic.py

```diff
@@ -0,0 +1,62 @@
+import json
+
+
+class Collection:
+    def __init__(self):
+        self.items = []
+        self.load_from_file()
+
+    def add_item(self, item):
+        self.items.append(item)
+        self.save_to_file()
+
+    def list_items(self):
+        for item in self.items:
+            for key, val in item.items():
+                print(f' {key} - {val} |', end=' ')
+            print(' ')
+
+    def edit_collection(self, item, key, val):
+        if key in item:
+            item[key] = val
+        self.save_to_file()
+
+    def search_title(self, title):
+        for item in self.items:
+            if item['title'] == title:
+                yield item
+
+    def search_director(self, director):
+        for item in self.items:
+            if item['director'] == director:
+                yield item
+
+    def search_year(self, year):
+        for item in self.items:
+            if item['year'] == year:
+                yield item
+
+    def search_genre(self, genre):
+        for item in self.items:
+            if item['genre'] == genre:
+                yield item
+
+    def save_to_file(self):
+        with open('df.json', 'w') as f:
+            json.dump(self.items, f)
+
+    def load_from_file(self):
+        try:
+            with open('df.json', 'r') as f:
+                self.items = json.load(f)
+        except FileNotFoundError:
+            pass

```

Comment:
there are two problems here:
1. your class is a good representation of a "god object", an object which can do anything, would be much better to separate some responsibilities, for example a class to work directly with a file, and a class to provide logic of editing and searching, hence the next point
2. The approach when you just read all the file content in the memory is not very good, you can at least do it only when some changes are required and when searching, it may take some additional time, but the memory will be free of the potentially huge file.